rust/src/dataset.rs

```diff
@@ -309,7 +309,8 @@ impl Dataset {
                 return Err(Error::from(batch.as_ref().unwrap_err()));
             }
         } else {
-            return Err(Error::EmptyDataset);
+            warn!("Dataset is empty, proceeding with empty schema");
+            schema = Schema::try_from(&ArrowSchema::empty())?;
```

Comment:
When creating an empty dataset, users should provide a schema.

I'd imagine that the semantics will be very similar to SQL `CREATE TABLE`.

Reply:
Yes indeed. Up until now we were extracting schema from the records but now that records might be absent in the case of empty dataset, schema must be provided or error will be thrown. This is what mostly this latest commit covers.